MantidQt/CustomInterfaces/src/Tomography/TomographyIfacePresenter.cpp

```diff
@@ -362,19 +528,24 @@ void TomographyIfacePresenter::processLogout() {
 }
 
 void TomographyIfacePresenter::processSetupReconTool() {
-  if (TomographyIfaceModel::g_CCPiTool != m_view->currentReconTool()) {
-    m_view->showToolConfig(m_view->currentReconTool());
-    m_model->updateReconToolsSettings(m_view->reconToolsSettings());
+  const std::string &currentReconTool = m_view->currentReconTool();
+  // save currentReconTool string to model
+  // retrieve, if the same, don't create a new Dialog pointer
+  // if not the same, create new Dialog
 
-    // TODO: this would make sense if the reconstruct action/button
-    // was only enabled after setting up at least one tool
-    // m_view->enableToolsSetupsActions(true);
+  auto dialog = m_configDialog.get();
+  if (!dialog) {
+    return;
```

Comment:
Yes this covers an edge case where on initialisation Qt calls the dialogue creation with an empty string, that call returns a nullptr dialogue and it's captured here.